src/components/UI/ErrorFallback.tsx

```diff
@@ -1,26 +1,29 @@
 import styled from '@emotion/styled';
 import PropTypes from 'prop-types';
-import React from 'react';
+import React, { ReactNode } from 'react';
 
 const ErrorWrapperSpan = styled.span`
   color: ${(props) => props.theme.colors.error};
   font-weight: 300;
 `;
 
+interface IErrorFallbackProps {
+  error: string;
+  children: ReactNode;
+}
+
 /**
  * This component will return the error passed to it,
  * if it exists, or the children, if it does not
  */
-function ErrorFallback({ errors, children, ...rest }) {
-  if (errors) {
-    return <ErrorWrapperSpan {...rest}>{errors}</ErrorWrapperSpan>;
-  }
+function ErrorFallback({ error, children }: IErrorFallbackProps) {
```

Comment:
This is usually typed with the React Function Component generic:
```typescript
const ErrorFallback: React.FC<IErrorFallbackProps> = ({ error, children }) => { ... }
```

Reply:
Thanks!

Reply:
I haven't found any exemples with function declarations instead of function expressions. Is there a way to do it with function declarations?

Reply:
That's because it isn't the "recommended" way of using it

I suppose something like `function ErrorFallback({ error, children }: IErrorFallbackProps): React.ReactElement` would work, but I haven't checked